src/dataExplorer/components/FluxQueryBuilder.tsx

```diff
@@ -3,36 +3,48 @@ import React, {FC, useState} from 'react'
 // Components
 import {DraggableResizer, Orientation} from '@influxdata/clockface'
 import {QueryProvider} from 'src/shared/contexts/query'
+import {EditorProvider} from 'src/shared/contexts/editor'
 import {ResultsProvider} from 'src/dataExplorer/components/ResultsContext'
 import ResultsPane from 'src/dataExplorer/components/ResultsPane'
+import SidePane from 'src/dataExplorer/components/SidePane'
 import Schema from 'src/dataExplorer/components/Schema'
 
 // Styles
 import './FluxQueryBuilder.scss'
 
-const INITIAL_VERT_RESIZER_HANDLE = 0.2
+const INITIAL_LEFT_VERT_RESIZER_HANDLE = 0.2
+const INITIAL_RIGHT_VERT_RESIZER_HANDLE = 0.8
 
 const FluxQueryBuilder: FC = () => {
   const [vertDragPosition, setVertDragPosition] = useState([
-    INITIAL_VERT_RESIZER_HANDLE,
+    INITIAL_LEFT_VERT_RESIZER_HANDLE,
+    INITIAL_RIGHT_VERT_RESIZER_HANDLE,
   ])
 
   return (
     <QueryProvider>
-      <DraggableResizer
-        handleOrientation={Orientation.Vertical}
-        handlePositions={vertDragPosition}
-        onChangePositions={setVertDragPosition}
-      >
-        <DraggableResizer.Panel>
-          <Schema />
-        </DraggableResizer.Panel>
-        <DraggableResizer.Panel className="new-data-explorer-rightside">
-          <ResultsProvider>
-            <ResultsPane />
-          </ResultsProvider>
-        </DraggableResizer.Panel>
-      </DraggableResizer>
+      <EditorProvider>
```

Comment:
having to set this so high just so other components can speak to it's injection mechanism seems to be a bit of code smell. lets say we have a button in the header that triggers the injection mechanism, do we now have to hoist the provider to the page level? does the meaning stay the same as the code floats in a different context, or is the connection between the disparate interactions constitute application logic which should be handled by a separate provider to cover the connection of these three components (fluxeditor, sidepane, schema)

Reply:
If you want the providers to be singletons, then they need to be singletons. Especially if it holds state (e.g. the connection to the editor).

Reply:
But if you have shared state to the editor access, then you really only want 1 connection to the editor. And the same provider where the editor is set via the callback from the react-monaco-editor.

Reply:
yeah no this isn't about the relationship between the editor context and the flux monaco editor, that one's ok, but about the relationship between the editor context and the other implementation specific components of the side panel and the schema browser. while i doubt there's going to be a product requirement to have a single injection mechanism update multiple editors, the code encapsulation should ensure that the module interdependence mechanism be resilient enough to handle an implementation change, especially since our next step is to experiment with this interaction. one proposed option explored by product is to have every button press in the schema browser call injection into a new flux query in an editor that is in preview mode, _then_ a separate button actually injects it into your working query. the interface as described in this pr is creating an implicit tight coupling between three things that should have little to do with each other and would make that option difficult to implement. there needs to be a controller layer here for encapsulating the application logic and decouple those components. Not expecting you to engineer for imaginary test cases here, just using examples to help illustrate that there are more flexible patterns to use in this case

Reply:
The provider holds state, the connection to the editor. Therefore we need a single connection to the editor, so a single provider used). The context use can be moved closer down the tree to the components, but the single connection/provider needs to be available to both the SidePane (with the FluxBrowser injecting functions) and the SchemaPane (to be injecting measurements), as well as the central pane (with the react-monaco-editor setting the editor instance).

So it's a single editor, a single editor connection, and a single injection point...via the common provider.

We don't want multiple connections to the editor, via multiple providers. (That is a recipe for race conditions etc.) We cannot entirely move our application away from any shared state...which is why we use the provider pattern for our shared state, with a single instance per provider. This matches our existing pattern and use of providers.

Reply:
I understand your point of implicit coupling. I am disagreeing with priorities for reducing bugs.

I am stating that a single src of connection to the editor instance is more important. And adding a pattern of event listeners (graph pattern into the editor) more messy and prone to bugs imho.

Reply:
i dont understand that jump? im not advocating for a particular control pattern, just that the components need to be decoupled to fill their application and provided a list of other cases where we broke from these implicit couplings through prop drilling in order to keep components reusable and maintainable.

Reply:
I got a greenlight to merge this for now, (to not block release of the FluxBroswer into the feature flagged new QX FluxBuilder). We can then have a conversation next week in followup about how we want to use providers in this part of the application.